frontend/src/components/submissionList.js

```diff
@@ -325,24 +328,33 @@ class SubmissionList extends React.Component {
       hasUsername: false,
       page: 1,
       pageSize:200,
-      pageCount: 1
+      pageCount: 1,
+      authEnabled: true
     }
   }
 
-  isAuthenticated() {
-    return this.props.userDetails &&
-        this.props.userDetails.hasOwnProperty('username') &&
-        this.props.userDetails.username !== null
-  }
-
   componentWillUnmount() {
     this.getSubmissionsPromise && this.getSubmissionsPromise.cancel();
     this.getFormDetailsPromise && this.getFormDetailsPromise.cancel();
+    this.getAuthStatusPromise && this.getAuthStatusPromise.cancel();
   }
 
   componentDidMount() {
     this.getSubmissions();
     this.getFormDetails();
+    if (!(this.props.userDetails &&
+        this.props.userDetails.hasOwnProperty('username') &&
+        this.props.userDetails.username !== null)) {
```

Comment:
Prefer `!= null` to also match `undefined` values.

Is `hasOwnProperty` needed here?